src/SWP/Bundle/ContentBundle/Loader/ArticleMediaLoader.php

```diff
@@ -0,0 +1,144 @@
+
+class ArticleMediaLoader implements LoaderInterface
+{
+    /**
+     * @var PublishWorkflowChecker
+     */
+    protected $publishWorkflowChecker;
+
+    /**
+     * @var DocumentManager
+     */
+    protected $dm;
+
+    /**
+     * @var string
+     */
+    protected $configurationPath;
+
+    /**
+     * @var CacheProvider
+     */
+    protected $metadataCache;
+
+    /**
+     * @var TenantAwarePathBuilderInterface
+     */
+    protected $pathBuilder;
+
+
+    /**
+     * @var MetaFactory
+     */
+    protected $metaFactory;
+
+    /**
+     * @var Context
+     */
+    protected $context;
+
+    /**
+     * ArticleMediaLoader constructor.
+     *
+     * @param PublishWorkflowChecker          $publishWorkflowChecker
+     * @param DocumentManager                 $dm
+     * @param                                 $configurationPath
+     * @param CacheProvider                   $metadataCache
+     * @param TenantAwarePathBuilderInterface $pathBuilder
+     * @param MetaFactory                     $metaFactory
+     * @param Context                         $context
+     */
+    public function __construct(
+        PublishWorkflowChecker $publishWorkflowChecker,
+        DocumentManager $dm,
+        $configurationPath,
+        CacheProvider $metadataCache,
+        TenantAwarePathBuilderInterface $pathBuilder,
+        MetaFactory $metaFactory,
+        Context $context
+    ) {
+        $this->publishWorkflowChecker = $publishWorkflowChecker;
+        $this->dm = $dm;
+        $this->configurationPath = $configurationPath.'/Resources/meta/media.yml';
+        $this->metadataCache = $metadataCache;
+        $this->pathBuilder = $pathBuilder;
+        $this->metaFactory = $metaFactory;
+        $this->context = $context;
+    }
+
+    /**
+     * Load meta object by provided type and parameters.
+     *
+     * @MetaLoaderDoc(
+     *     description="Article Loader loads articles from Content Repository",
+     *     parameters={
+     *         contentPath="SINGLE|required content path",
+     *         slug="SINGLE|required content slug",
+     *         pageName="COLLECTION|name of Page for required articles"
+     *     }
+     * )
+     *
+     * @param string $type         object type
+     * @param array  $parameters   parameters needed to load required object type
+     * @param int    $responseType response type: single meta (LoaderInterface::SINGLE) or collection of metas (LoaderInterface::COLLECTION)
+     *
+     * @return Meta|Meta[]|bool false if meta cannot be loaded, a Meta instance otherwise
+     */
+    public function load($type, $parameters = array(), $responseType = LoaderInterface::SINGLE)
+    {
+        if ($responseType === LoaderInterface::COLLECTION) {
+            $media = false;
+            if (is_array($parameters) && array_key_exists('article', $parameters)) {
+                $media = $this->dm->find(null, $parameters['article']->getValues()->getId().'/media');
```

Comment:
It's not configurable for now - just hardcoded so it can't break. But saving it as constant in `ArticleMedia` sounds like a plan.